pq-crypto/sike_r2/sike_r2_kem.c

```diff
@@ -110,9 +110,9 @@ int SIKE_P434_r2_crypto_kem_dec(unsigned char *ss, const unsigned char *ct, cons
 
     // Generate shared secret ss <- H(m||ct) or output ss <- H(s||ct)
     EphemeralKeyGeneration_A(ephemeralsk_.d, c0_);
-    if (memcmp(c0_, ct, CRYPTO_PUBLICKEYBYTES) != 0) {
-        memcpy(temp, sk, MSG_BYTES);
-    }
+    bool copy_or_dont = s2n_constant_time_equals(c0_, ct, CRYPTO_PUBLICKEYBYTES);
+    s2n_constant_time_copy_or_dont(temp, sk, MSG_BYTES, copy_or_dont);
```

Comment:
It took me several times to read through this (including the old, the new, and the implementations of the called function) to be think that it is correct. While this code is likely correct, I think we need to improve our comments to make it clearer.

```suggestion
    // We only want to copy the data if c0_ and ct are different
    bool dont_copy = s2n_constant_time_equals(c0_, ct, CRYPTO_PUBLICKEYBYTES);
    // The last argument to s2n_constant_time_copy_or_dont is dont and thus prevents the copy when non-zero/true
    s2n_constant_time_copy_or_dont(temp, sk, MSG_BYTES, dont_copy);
```

If my comments are wrong, then the code is probably wrong too.